### PR TITLE
heynote: init at 1.6.0

### LINKED ISSUE #277545: Package request: heynote
**Project description**

A dedicated scratchpad for developers

**Metadata**

* homepage URL: https://heynote.com/
* source URL: https://github.com/heyman/heynote/
* license: MIT
* platforms: Mac, Windows, and Linux.
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Ligthiago**
## Description of changes

New package: heynote
fixes https://github.com/NixOS/nixpkgs/issues/277545

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/he/heynote/package.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    runHook preInstall
+    
+    mkdir -p  $out/bin $out/share/${pname} $out/share/applications $out/share/pixmaps
+    cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
+    cp -a ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
+    cp -a ${appimageContents}/usr/share/icons $out/share
+    ln -s $out/share/icons/hicolor/0x0/apps/${pname}.png $out/share/pixmaps/${pname}.png
+
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
```
Comment (dotlambda) [inline pkgs/by-name/he/heynote/package.nix:38]: Please replace `${pname}` with `heynote`. See #277994.

File: pkgs/by-name/he/heynote/package.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  meta = with lib; {
+    description = "A dedicated scratchpad for developers";
+    homepage = "https://github.com/heyman/heynote";
+    changelog = "https://github.com/heyman/heynote/releases/tag/v${version}";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ Ligthiago ];
+    mainProgram = "heynote";
+  };
+}
```
Comment (dotlambda) [inline pkgs/by-name/he/heynote/package.nix:58]:
```suggestion
}

```

File: pkgs/by-name/he/heynote/package.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    runHook preInstall
+
+    mkdir -p  $out/bin $out/share/heynote $out/share/applications $out/share/pixmaps
+    cp -a ${appimageContents}/{locales,resources} $out/share/heynote
+    cp -a ${appimageContents}/heynote.desktop $out/share/applications/heynote.desktop
+    cp -a ${appimageContents}/usr/share/icons $out/share
+    ln -s $out/share/icons/hicolor/0x0/apps/heynote.png $out/share/pixmaps/heynote.png
+
+    substituteInPlace $out/share/applications/heynote.desktop \
+      --replace 'Exec=AppRun' 'Exec=heynote'
```
Comment (dotlambda) [inline pkgs/by-name/he/heynote/package.nix:38]:
```suggestion
      --replace-fail 'Exec=AppRun' 'Exec=heynote'
```

File: pkgs/by-name/he/heynote/package.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p  $out/bin $out/share/heynote $out/share/applications $out/share/pixmaps
+    cp -a ${appimageContents}/{locales,resources} $out/share/heynote
+    cp -a ${appimageContents}/heynote.desktop $out/share/applications/heynote.desktop
+    cp -a ${appimageContents}/usr/share/icons $out/share
```
Comment (dotlambda) [inline pkgs/by-name/he/heynote/package.nix:34]: Can we use `install`?

File: pkgs/by-name/he/heynote/package.nix [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    ln -s $out/share/icons/hicolor/0x0/apps/heynote.png $out/share/pixmaps/heynote.png
+
+    substituteInPlace $out/share/applications/heynote.desktop \
+      --replace 'Exec=AppRun' 'Exec=heynote'
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper ${electron}/bin/electron $out/bin/heynote \
```
Comment (dotlambda) [inline pkgs/by-name/he/heynote/package.nix:44]:
```suggestion
    makeWrapper ${lib.getExe electron} $out/bin/heynote \
```

**Review by dotlambda — COMMENTED** [review]
Please set `meta.sourceProvenance`.

**Comment by dotlambda** [comment]
Did you try building from source? If not have a look at other packages using electron-builder. It shouldn't be too hard.

**Review by FliegendeWurst — APPROVED** [review]
Builds fine. Note: 1.8.0 was released

**Pull request closed without merging**